### PR TITLE
fix(data-export): Make query a required field

### PR DESCRIPTION
**Pull request by wmak**
- Fixes SENTRY-PPJ
- This fixes a bug where we would 500 when query wasn't included in the
  payload because the DiscoverProcessor assumes that query will always
  be available
- Could default to empty string, but I think we should ask the user to
  be explicit instead of making that guess for them

### CODE REVIEW

File: src/sentry/data_export/endpoints/data_export.py [hunk 1]
```diff
@@ -51,6 +51,10 @@ def validate(self, data):
                 detail = f"You can export up to {MAX_FIELDS} fields at a time. Please delete some and try again."
                 raise serializers.ValidationError(detail)
 
+            if "query" not in query_info:
```
Comment (Zylphrex) [inline src/sentry/data_export/endpoints/data_export.py:54]: Can we add an additional condition that `fields` is an non-empty array? Can be another PR though.

**Review by Zylphrex — APPROVED** [review]

**Review by dashed — APPROVED** [review]
